code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -157,8 +160,72 @@
 
 	modes = list(MODE_NONE = MODE_SHUTTLE, MODE_SHUTTLE = MODE_NONE)
 
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging
+	name = "atmospheric thermal imaging goggles"
+	desc = "Goggles used by Atmospheric Technician to see the thermal energy of gasses in open areas"
+	icon_state = "trayson-radiation"
+	inhand_icon_state = "trayson-meson"
+	glass_colour_type = /datum/client_colour/glass_colour/gray
+
+	modes = list(MODE_NONE = MODE_ATMOS_THERMAL, MODE_ATMOS_THERMAL = MODE_NONE)
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/toggle_mode(mob/user, voluntary)
+	mode = modes[mode]
+	to_chat(user, "<span class='[voluntary ? "notice":"warning"]'>[voluntary ? "You turn the goggles":"The goggles turn"] [mode ? "to [mode] mode":"off"][voluntary ? ".":"!"]</span>")
+	switch(mode)
+		if(MODE_ATMOS_THERMAL)
+			change_glass_color(user, /datum/client_colour/glass_colour/lightorange)
+		if(MODE_NONE)
+			change_glass_color(user, initial(glass_colour_type))
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.glasses == src)
+			H.update_sight()
+
+	update_appearance()
+	update_action_buttons()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/update_icon_state()
+	icon_state = inhand_icon_state = "trayson-[mode]"
+	return ..()
+
+/obj/item/clothing/glasses/meson/engine/atmos_imaging/process()
+	if(!ishuman(loc))
+		return
+	var/mob/living/carbon/human/user = loc
+	if(user.glasses != src || !user.client)
+		return
+	if(mode == MODE_ATMOS_THERMAL)
+		atmos_thermal(user)
+
+/proc/atmos_thermal(mob/viewer, range = 4, duration = 10)
+	if(!ismob(viewer) || !viewer.client)
+		return
+	for(var/turf/open in view(range, viewer))
+		if(open.blocks_air)
+			continue
+		var/datum/gas_mixture/environment = open.return_air()
+		var/temp = round(environment.temperature)
+		var/mutable_appearance/temptext = new()
+		temptext.icon = 'icons/turf/overlays.dmi'
+		temptext.plane = RUNECHAT_PLANE
+		temptext.alpha = 200
+		temptext.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+		var/image/pic = image(loc = open)
+		if(temp <= 273.15)
+			temptext.icon_state = "blueOverlay"
+		else if(temp >= 303.15)
+			temptext.icon_state = "redOverlay"
+		else 
+			temptext.icon_state = "greenOverlay"
```

Comment:
So the overlay colors are hard-sprited?

Reply:
sadly yes, i do not know how to make it changes color based on temp, the heat2color proc only works with higher temp. So for now i reckon it would just be hard sprited for above 30 C and below 0 C

Reply:
shouldn't you use temperature defines and pick the thresholds for humans to be harmed by the temperature